alf/examples/metadrive/ppg_metadrive_transformer_conf.py

```diff
@@ -61,57 +80,85 @@ class ObservationCombiner(torch.nn.Module):
+        self._ego_eb = ego_feature.make_embedding_net(d_model)
 
     def forward(self, inputs):
         map_sequence, ego, agents = inputs
 
-        x0 = self._ego_fc(ego).unsqueeze(1)  # [B, 1, d_model]
+        x0 = self._ego_eb(ego).unsqueeze(1)  # [B, 1, d_model]
 
         # The input ``sequence`` is [B, L, d_map_feature]
-        x1 = self._map_fc(map_sequence)  # [B, L, d_model]
+        x1 = self._map_eb(map_sequence)  # [B, L, d_model]
 
         x2 = agents.view(*agents.shape[:2], -1)
-        x2 = self._agent_fc(x2)  # [B, A, d_model]
+        x2 = self._agent_eb(x2)  # [B, A, d_model]
 
         return torch.cat([x0, x1, x2], dim=1)
 
 
+class MaskedTransformer(torch.nn.Module):
+    def __init__(self, d_model, num_heads, num_layers, map_limit, agent_limit):
+        super().__init__()
+        self._memory_size = 1 + map_limit + agent_limit
+
+        tf_layers = []
+        for i in range(num_layers):
+            tf_layers.append(
+                alf.layers.TransformerBlock(
+                    d_model=d_model,
+                    num_heads=num_heads,
+                    memory_size=self._memory_size,
+                    positional_encoding='none'))
+        self._tf_layers = torch.nn.ModuleList(tf_layers)
+
+    def forward(self, inputs):
+        x, map_mask, agent_mask = inputs
+        B = x.shape[0]
+        mask = torch.hstack((torch.ones(B, 1, dtype=bool), map_mask,
+                             agent_mask))
+        for layer in self._tf_layers:
+            x = layer(memory=x, mask=mask)
+
+        return x
+
+
 def encoding_network_ctor(input_tensor_spec):
     d_model = 128
     num_heads = 8
-    memory_size = (input_tensor_spec['map'].shape[0] +
-                   input_tensor_spec['agents'].shape[0] + 1)
 
     layers = [
-        lambda x: (x['map'], x['ego'], x['agents']),
-        ObservationCombiner(
-            d_map_feature=input_tensor_spec['map'].shape[-1],
-            d_ego_feature=input_tensor_spec['ego'].shape[-1],
-            d_agent_feature=input_tensor_spec['agents'].shape[1] *
-            input_tensor_spec['agents'].shape[2],
-            d_model=d_model),
+        alf.nn.Branch(
+            alf.nn.Sequential(
+                lambda x: (x['map'], x['ego'], x['agents']),
+                ObservationCombiner(
+                    map_feature=EmbeddingConfig(
+                        d_input=input_tensor_spec['map'].shape[-1]),
+                    ego_feature=EmbeddingConfig(
+                        d_input=input_tensor_spec['ego'].shape[-1],
+                        hidden=(32, )),
+                    agent_feature=EmbeddingConfig(
+                        d_input=input_tensor_spec['agents'].shape[1] *
+                        input_tensor_spec['agents'].shape[2],
+                        hidden=(32, )),
+                    d_model=d_model),
+                input_tensor_spec=input_tensor_spec), lambda x:
+            (x['map_mask'], x['agent_mask'])), lambda x: (x[0], x[1][0], x[1][
```

Comment:
This auto-formatting makes it quite hard to read. 
The lambda at line 151 should be indented at the same level ax ObservationCombiner.
The lambda at line 152 should be at a new line and be indented at the same level as MaskedTransformer.
Perhaps use the following to disable the auto-formatting?
```
# yapf: diasble
...
# yapf: enable
```

Reply:
Right, thanks for the advice. Updated.